pkg/common/cns-lib/volume/manager.go

```diff
@@ -2443,6 +2452,19 @@ func (m *defaultManager) createSnapshotWithImprovedIdempotencyCheck(ctx context.
 	// Handle snapshot operation result
 	createSnapshotsOperationRes := createSnapshotsTaskResult.GetCnsVolumeOperationResult()
 	if createSnapshotsOperationRes.Fault != nil {
+		err = soap.WrapVimFault(createSnapshotsOperationRes.Fault.Fault)
+		if cnsvsphere.IsCnsSnapshotCreatedFaultError(err) {
+			errMsg := fmt.Sprintf("snapshot %q on volume %q got created, but post-processing failed. "+
+				"Fault: %q, opID: %q", instanceName, volumeID, spew.Sdump(createSnapshotsOperationRes.Fault),
+				createSnapshotsTaskInfo.ActivationId)
+
+			snapshotId := createSnapshotsOperationRes.Fault.Fault.(cnstypes.CnsSnapshotCreatedFault).SnapshotId.Id
+			volumeOperationDetails = createRequestDetails(instanceName, volumeID, snapshotId, 0, nil,
+				volumeOperationDetails.OperationDetails.TaskInvocationTimestamp, createSnapshotsTask.Reference().Value,
+				"", createSnapshotsTaskInfo.ActivationId, taskInvocationStatusPartiallyFailed, errMsg)
+			return nil, logger.LogNewError(log, errMsg)
+		}
+
```

Comment:
Should this be within the "if m.idempotencyHandlingEnabled" block starting at line 2472?

Reply:
moved this code within "if m.idempotencyHandlingEnabled" block